### PR TITLE
Fix bugs in StdoutJsonParser when a line spanned multiple calls to appendOutput.

### PR DESCRIPTION
**Pull request by jacob314**
The downside of this approach is we will not display a line until a \n occurs but that appears to be fine.

An alternate robust approach would be to use an actual streaming JSON parser to track when the JSON blocks are complete.

The previous approach failed if the JSON had an }] too early in the block and failed on json responses larger than a single call to appendOutput.

### CODE REVIEW

**Comment by jacob314** [comment]
Added tests covering the failure.
Also removed flush() from the API as it is not needed now that we always flush as long as we aren't in a suspected `[{` JSON block.
Verified that this doesn't create extra new lines and does cause text to display even before the first line feed is issued.

File: src/io/flutter/utils/StdoutJsonParser.java [hunk 1]
```diff
@@ -10,59 +10,50 @@
 
 /**
  * A class to process regular text output intermixed with newline-delimited JSON.
+ *
+ * JSON lines starting with [{ are never be split into multiple lines even
```
Comment (devoncarew) [inline src/io/flutter/utils/StdoutJsonParser.java]: ==> `are never split`

File: src/io/flutter/utils/StdoutJsonParser.java [hunk 2]
```diff
@@ -10,59 +10,50 @@
...
+ * if they are emitted over the course of multiple calls to appendOutput.
+ * Regular lines on the other hand are emitted immediately so that debug output
+ * always shows up when expected..
```
Comment (devoncarew) [inline src/io/flutter/utils/StdoutJsonParser.java]: fewer periods :)

File: testSrc/unit/io/flutter/utils/StdoutJsonParserTest.java [hunk 3]
```diff
@@ -58,12 +57,43 @@ public void split_json() throws Exception {
...
 
     assertArrayEquals(
       "validating parser results",
       new String[]{"hello\n", "there\n", "[{'foo':'bar'}]\n", "bye\n"},
       parser.getAvailableLines().toArray()
     );
   }
+
+  @Test
+  public void deep_nested_json() throws Exception {
```
Comment (devoncarew) [inline testSrc/unit/io/flutter/utils/StdoutJsonParserTest.java]: nit: I believe our test names use lowerCamelCase

File: src/io/flutter/utils/StdoutJsonParser.java (hunk 1 above)
Comment (jacob314) [reply src/io/flutter/utils/StdoutJsonParser.java]: done

File: src/io/flutter/utils/StdoutJsonParser.java (hunk 2 above)
Comment (jacob314) [reply src/io/flutter/utils/StdoutJsonParser.java]: done..........

File: testSrc/unit/io/flutter/utils/StdoutJsonParserTest.java (hunk 3 above)
Comment (jacob314) [reply testSrc/unit/io/flutter/utils/StdoutJsonParserTest.java]: done. for some reason the existing tests in this file used_under_scores_so_i_copied_them :)

**Pull request merged**